models/_sector/nft/trades/ethereum/platforms/collectionswap_ethereum_base_trades.sql

```diff
@@ -28,54 +28,62 @@ raw_trades as (
+                ,cast(cardinality(nftIds) as uint256) as number_of_items
+                ,cast(outputAmount as uint256) as price_raw
+                ,cast(protocolFee as uint256) as platform_fee_amount_raw
+                ,cast(JSON_EXTRACT_SCALAR(royaltyDue[1], '$.amount') as uint256) as royalty_fee_amount_raw
+                ,from_hex(JSON_EXTRACT_SCALAR(royaltyDue[1], '$.recipient')) as royalty_fee_address
+                ,cast(tradeFee as uint256) as trade_fee_amount_raw
                 ,contract_address as project_contract_address
             from {{ source('collectionswap_ethereum','CollectionPool_evt_SwapNFTOutPool') }} e
             {% if is_incremental() %}
-            WHERE evt_block_time >= date_trunc("day", now() - interval '1 week')
+            WHERE evt_block_time >= date_trunc('day', now() - interval '7' day)
             {% else %}
-            WHERE evt_block_time >= '{{project_start_date}}'
+            WHERE evt_block_time >= {{project_start_date}}
             {% endif %}
             union all
             select
-                evt_block_number as block_number
-                ,evt_block_time as block_time
-                ,evt_index
-                ,evt_tx_hash as tx_hash
-                ,contract_address as buyer
-                ,null as seller
-                ,'Sell' as trade_category
-                ,nftIds as nft_id_array
-                ,cardinality(nftIds) as number_of_items
-                ,cast(inputAmount + protocolFee + cast(get_json_object(royaltyDue[0], '$.amount') as decimal(38)) as decimal(38)) as price_raw
-                ,cast(protocolFee as decimal(38)) as platform_fee_amount_raw
-                ,get_json_object(royaltyDue[0], '$.amount') as royalty_fee_amount_raw
-                ,get_json_object(royaltyDue[0], '$.recipient') as royalty_fee_address
-                ,cast(tradeFee as decimal(38)) as trade_fee_amount_raw
-                ,contract_address as project_contract_address
+                evt_block_number AS block_number,
+                evt_block_time AS block_time,
+                evt_index,
+                evt_tx_hash AS tx_hash,
+                contract_address AS buyer,
+                cast(null as varbinary) AS seller,
+                'Sell' AS trade_category,
+                nftIds AS nft_id_array,
+                cast(cardinality(nftIds) as uint256) AS number_of_items,
+                TRY_CAST(
+                inputAmount + protocolFee + TRY_CAST(
+                  JSON_EXTRACT_SCALAR(royaltyDue[1], '$.amount') AS UINT256
+                ) AS UINT256
+                ) AS price_raw,
+                TRY_CAST(protocolFee AS UINT256) AS platform_fee_amount_raw,
+                TRY_CAST(JSON_EXTRACT_SCALAR(royaltyDue[1], '$.amount')  AS UINT256) AS royalty_fee_amount_raw,
+                from_hex(JSON_EXTRACT_SCALAR(royaltyDue[1], '$.recipient')) AS royalty_fee_address,
+                TRY_CAST(tradeFee AS UINT256) AS trade_fee_amount_raw,
```

Comment:
Why is the TRY_CAST required?

Reply:
Don't think it's required here, but a remnant from using the harmonizer which defaults to `TRY_CAST`